### PR TITLE
Authentication through QR code

### PR DESCRIPTION
**Pull request by VincentCATILLON**
## Detailed purpose of the PR

This PR adds authentication through QR code with permissions handler.

Use this QR code to authenticate:
capture d ecran 2019-03-08 a 17 01 03

(Source: [https://user-images.githubusercontent.com/15608581/53570829-bbcdae80-3b67-11e9-80c9-12775f07422c.png](https://user-images.githubusercontent.com/15608581/53570829-bbcdae80-3b67-11e9-80c9-12775f07422c.png))

### Result and observation

You can authenticate use QR code to authenticate 

- [ ] **Breaking changes ?**
- [ ] **Extra dependency ?**

### Testing Strategy

- [ ] Already covered by tests
- [ ] Manual testing
- [ ] Unit testing

### CODE REVIEW

File: src/screens/authentication.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  ...ReactNavigation$ScreenProps
+  // ...ConnectedDispatchProps
+|};
+
+class AuthenticationScreen extends React.PureComponent<Props> {
+  props: Props;
+
+  handleScan = (token?: string) => {
+    const {navigation} = this.props;
+    // this.props.signIn(token);
```
Comment (VincentCATILLON) [inline src/screens/authentication.js]: @Fausto95 this is what you should trigger once the action is ready (cc. @godu)

File: src/containers/with-permissions.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        appState === APP_STATE.ACTIVE
+      ) {
+        this.checkPermissions();
+      }
+      this.setState({
+        appState
+      });
+    };
+
+    checkPermissions = () => types.forEach(type => this.props.checkPermission(type));
```
Comment (Fausto95) [inline src/containers/with-permissions.js]: Where is `types` coming from?

**Comment by VincentCATILLON** [comment]
@Fausto95 Don't forget to test the authentication screen integration on iPhone X (with noche and safe area). We have a prop available on `screen` component since few days: `noSafeArea`, IMO we should use it to have the background gradient color on the entire screen (top to bottom)

File: src/containers/with-permissions.js (hunk 2 above)
Comment (VincentCATILLON) [reply src/containers/with-permissions.js]: from HOC params: https://github.com/CoorpAcademy/mobile/pull/146/files/34e9e86a937ea7e0353554db67d1225713ab5670#diff-8189e074f90bada4a923f1a8ebc9cb52R19

Comment (VincentCATILLON) [reply src/containers/with-permissions.js]: And used like this:
https://github.com/CoorpAcademy/mobile/pull/146/files/34e9e86a937ea7e0353554db67d1225713ab5670#diff-c3ceacf5c0e83626ab95dda522c88b5aR92

File: __e2e__/context.test.js [hunk 3]
```diff
@@ -21,6 +23,8 @@ describe('Context', () => {
     });
 
     it('should go to the question', async () => {
+      await waitFor(element(by.id('goToQuestion-button'))).toBeVisible();
```
Comment (chrisdugne) [inline __e2e__/context.test.js]: not required

File: __e2e__/utils.js [hunk 4]
```diff
@@ -2,24 +2,60 @@
...
-const getContextTab = el => el(by.id('slide-tab')).atIndex(3);
-const getQuestionTab = el => el(by.id('slide-tab')).atIndex(2);
-const getLessonTab = el => el(by.id('slide-tab')).atIndex(1);
-const getClueTab = el => el(by.id('slide-tab')).atIndex(0);
+const getContextTab = (el: DetoxElement) => el(by.id('slide-tab')).atIndex(3);
+const getQuestionTab = (el: DetoxElement) => el(by.id('slide-tab')).atIndex(2);
+const getLessonTab = (el: DetoxElement) => el(by.id('slide-tab')).atIndex(1);
+const getClueTab = (el: DetoxElement) => el(by.id('slide-tab')).atIndex(0);
+
+const byPassAuthentication = async () => {
```
Comment (chrisdugne) [inline __e2e__/utils.js]: it's rather `bypassAuthentication`

File: scripts/pull-translations.js [hunk 5]
```diff
@@ -78,16 +91,42 @@ const generate = async (locale: string) => {
...
+    permission: 'Permission',
+    // @todo get it from github
+    permissionCamera:
+      'We need access to your camera in order to scan your QR code and authenticate you.',
     question: playerTranslations.Question,
+    quit: playerTranslations.Quit,
     retryLevel: playerTranslations['Retry level'],
-    selectSomethingBelow: playerTranslations['Select something below'],
+    // @todo get it from github
+    loginButton: playerTranslations['Scan your QR code'] || '',
```
Comment (alanlanglois) [inline scripts/pull-translations.js]: Would be better to keep alphabetical order for easier readability

**Pull request merged**